devicetypes/smartthings/zigbee-multi-button.src/zigbee-multi-button.groovy

```diff
@@ -0,0 +1,208 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ *	Author: SRPOL
+ *	Date: 2019-02-18
+ */
+
+import groovy.json.JsonOutput
+import physicalgraph.zigbee.zcl.DataType
+
+metadata {
+	definition (name: "Zigbee Multi Button", namespace: "smartthings", author: "SmartThings") {
+		capability "Actuator"
+		capability "Battery"
+		capability "Button"
+		capability "Holdable Button"
+		capability "Configuration"
+		capability "Refresh"
+		capability "Sensor"
+		capability "Health Check"
+
+		fingerprint inClusters: "0000, 0001, 0003, 0007, 0020, 0B05", outClusters: "0003, 0006, 0019", manufacturer: "CentraLite", model:"3450-L", deviceJoinName: "Iris KeyFob 1"
+	}
+
+	tiles {
+		standardTile("button", "device.button", width: 2, height: 2) {
+			state "default", label: "", icon: "st.unknown.zwave.remote-controller", backgroundColor: "#ffffff"
+			state "button 1 pushed", label: "pushed #1", icon: "st.unknown.zwave.remote-controller", backgroundColor: "#00A0DC"
+		}
+
+		valueTile("battery", "device.battery", decoration: "flat", inactiveLabel: false) {
+			state "battery", label:'${currentValue}% battery', unit:""
+		}
+
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat") {
+			state "default", action:"refresh.refresh", icon:"st.secondary.refresh"
+		}
+		main (["button"])
+		details(["button", "battery", "refresh"])
+	}
+}
+
+def parse(String description) {
+	def map = zigbee.getEvent(description)
+	def result = map ? map : parseAttrMessage(description)
+	log.debug "Description ${description} parsed to ${result}"
+	return result
+}
+
+def parseAttrMessage(description) {
+	def descMap = zigbee.parseDescriptionAsMap(description)
+	def map = [:]
+	if (descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.commandInt != 0x07 && descMap?.value) {
+		map = getBatteryPercentageResult(Integer.parseInt(descMap.value, 16))
+	} else if (descMap?.clusterInt == zigbee.ONOFF_CLUSTER && descMap.isClusterSpecific) {
+		map = getButtonEvent(descMap)
+	}
+	map
+}
+
+def getButtonEvent(descMap) {
+	if (descMap.commandInt == 1) {
+		getButtonResult("press")
+	}
+	else if (descMap.commandInt == 0) {
+		def button = buttonMap[device.getDataValue("model")][descMap.sourceEndpoint]
+		getButtonResult("release", button)
+	}
+}
+
+def getButtonResult(buttonState, buttonNumber = 1) {
+	def event = [:]
+	if (buttonState == 'release') {
+		def timeDiff = now() - state.pressTime
+		if (timeDiff > 10000) {
+			return event
+		} else {
+			buttonState = timeDiff < holdTime ? "pushed" : "held"
+			def descriptionText = "${device.displayName[0..-2]} button $buttonNumber was $buttonState"
```

Comment:
Is the display name not set by the user?  I don't think we should arbitrarily be stripping off characters.

Reply:
This is the same pattern we used for aeon minimote. It saves us the headache of having an "empty" device in our apps.

Zach's write about the displayname, though. You can't guarantee it will always have a number at the end. This is just for a log message, though, so I don't think it's too dire.